### ISSUE TITLE
Strapi crashing with CKEditor

### ISSUE THREAD

**Issue by OssiPesonen** [strapi/strapi#8281]
**Describe the bug**
Every time I input a certain type of content to CKEditor, such as `...`, Strapi content creation page freezes completely.

**Steps to reproduce the behavior**

1. Clone https://github.com/OssiPesonen/ossi-dev-strapi
2. Run `npm install` etc. to get the app running. Set up your credentials etc. and log in
3. Go to Collection Types and open up any content type. Click **Add new*** 
4. Write `...` in to the editor and see your browser freeze

**Expected behavior**
The website should not freeze.

**Screenshots**

**Code snippets**
https://github.com/OssiPesonen/ossi-dev-strapi

**System**

- Node.js version: 12.19.0
- NPM version: 6.14.6
- Strapi version: 3.2.1
- Database: MySQL
- Operating system: Debian / Mac OS X / Windows (run it on all systems)

**Additional context**
I do apologize if this is not your fault, but I've run the **exact same CKEditor custom build** on an internal documentation application (Next.js + Contentful) and there are no such problems there. I've run into numerous problems when I switched DraftJS to CKeditor including, but not limited to:

- Base64 image insert magically stopped working
- I had to disable `autoformat` because it also kept freezing.
- ClassicEditor sticky doesn't work but that's not Strapi's fault
- MediaLib always inserts images at the bottom of the editor, but once again that's not Strapi's fault

Unfortunately because the entire page freezes I get no error logs of any kind. I've tried numerous builds after this initial one but they all end up with the same problem and I can't understand why.

**Comment by soupette** [comment]
Hello @OssiPesonen 
I have checked your code and I don't understand why do you need an effect in this [file](https://github.com/OssiPesonen/ossi-dev-strapi/blob/f1d4235b658550f82acf76920bd62da96ae905f8/extensions/content-manager/admin/src/components/CKEditor/index.js#L56-L60)

Try to update your code as follows, so you remove the infinite loop that is causing your app to crash:

```
const Editor = ({ onChange, name, value }) => {
  const { CKEditor, ClassicEditor, isEditorLoaded } = useCKEditor();

  return (
    <Wrapper className="ck-editor">
      {isEditorLoaded && (
        <CKEditor
          config={editorConfig}
          editor={ClassicEditor}
          data={value}
          onChange={(event, editor) => {
            const data = editor.getData();
            onChange({ target: { name, value: data } });
          }}
        />
      )}
    </Wrapper>
  );
};
```

Also, I am not sure you need your hook to load dynamically your component you can use `React.lazy`

**Comment by OssiPesonen** [comment]
Thank you so much for pointing an issue with my code! I should not have opened an issue on the repo, as this was not Strapi's fault, but rather opened up a discussion. 

I checked my code and indeed it was the `useEffect` that caused an infinite loop because. I used that because in previous development I had to wait until `value` was available from the API and only then could I initialize CKEditor. It has something to do with the editor not initializing properly if you insert an existing value AFTER the editor has been initialized, but I cannot remember why I created the hook exactly. I'm not too familiar with React.lazy, but I'll look into it! 

I am closing this issue. Thanks again!
